File: citizen app/backend/api/app/schemas/citizen.py
"""`GET /api/citizen/home` — one optimized dashboard payload (section 42)
instead of the mobile app firing five separate requests on launch."""
from pydantic import BaseModel

from app.schemas.hazard import HazardOut


class HomeStats(BaseModel):
    nearby_count: int
    critical_count: int
    warning_count: int


class HomeResponse(BaseModel):
    greeting: str
    user_name: str
    city_name: str
    stats: HomeStats
    nearby_hazards: list[HazardOut]
    map_markers: list[HazardOut]

File: citizen_app/apps/citizen-mobile/types/citizen.ts
import type { Hazard } from './hazard';

export interface HomeStats {
  nearbyCount: number;
  criticalCount: number;
  warningCount: number;
}

export interface HomeResponse {
  greeting: string;
  userName: string;
  cityName: string;
  stats: HomeStats;
  nearbyHazards: Hazard[];
  mapMarkers: Hazard[];
}
